api/cas/views.py

```diff
@@ -0,0 +1,310 @@
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        body_data = self.load_request_body_data(request)
+        account_action = body_data.get('accountAction')
+
+        if account_action != 'PASSWORD_FORGOT':
+            raise ValidationError(detail=messages.INVALID_REQUEST)
+
+        account.handle_password_forgot(body_data.get('user'))
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class AccountPasswordReset(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Reset the password for an osf account.
+    """
+
+    view_category = 'cas'
+    view_name = 'account-password-reset'
+    permission_classes = ()
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        body_data = self.load_request_body_data(request)
+        account_action = body_data.get('accountAction')
+
+        if account_action != 'PASSWORD_RESET':
+            raise ValidationError(detail=messages.INVALID_REQUEST)
+
+        user = account.handle_password_reset(body_data.get('user'))
+
+        content = {
+            'verificationKey': user.verification_key,
+            'userId': user._id,
+            'casAction': 'account-password-reset',
+            'nextUrl': False,
+        }
+
+        return Response(data=content, status=status.HTTP_200_OK)
+
+
+class ServiceOauthToken(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Get the owner and scopes of a personal access token by token id.
+    """
+
+    view_category = 'cas'
+    view_name = 'service-oauth-token'
+    permission_classes = ()
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        body_data = self.load_request_body_data(request)
+        content = service.get_oauth_token(body_data)
+        return Response(data=content, status=status.HTTP_200_OK)
+
+
+class ServiceOauthScope(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Get the description of the oauth scope by scope name.
+    """
+
+    view_category = 'cas'
+    view_name = 'service-oauth-scope'
+    permission_classes = ()
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        body_data = self.load_request_body_data(request)
+        content = service.get_oauth_scope(body_data)
+        return Response(data=content, status=status.HTTP_200_OK)
+
+
+class ServiceOauthApps(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Load all active developer applications.
+    """
+
+    view_category = 'cas'
+    view_name = 'service-oauth-apps'
+    permission_classes = ()
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        body_data = self.load_request_body_data(request)
+        content = service.load_oauth_apps(body_data)
+        return Response(data=content, status=status.HTTP_200_OK)
+
+
+class ServiceInstitutions(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
```

Comment:
Load all active institutions that support login. CAS backend periodically (every 2 minutes) makes this API request to update the list.